### ISSUE TITLE
Conformer FastSpeech2 fine-tuning suspended at 24epoch

### ISSUE THREAD

**Issue by zzxiang** [espnet/espnet#3721]
I'm fine-tuning a conformer_fastspeech2_tacotron2 model on a custom data with `g2p=pyopenjtalk_prosody`. I used [egs2/jsut/tts1/conf/tuning/train_conformer_fastspeech2.yaml](https://github.com/espnet/espnet/blob/master/egs2/jsut/tts1/conf/tuning/train_conformer_fastspeech2.yaml) but modified `lr` to 0.1.

The training of tacotron2 has already completed. But while I trained conformer_fastspeech2, the training just suspended on 24epoch and didn't move forward.

Here's the end of train.log:

```
[spsy-ml-2] 2021-10-30 16:52:21,022 (trainer:653) INFO: 24epoch:train:901-950batch: iter_time=2.215e-04, forward_time=0.312, l1_loss=0.428, duration_loss=0.016, pitch_loss=0.127, energy_loss=0.021, loss=0.592, backward_time=0.433, optim_step_time=0.025, optim0_lr0=4.829e-05, train_time=8.137
[spsy-ml-2] 2021-10-30 16:53:01,763 (trainer:653) INFO: 24epoch:train:951-1000batch: iter_time=2.622e-04, forward_time=0.311, l1_loss=0.428, duration_loss=0.016, pitch_loss=0.126, energy_loss=0.020, loss=0.590, backward_time=0.434, optim_step_time=0.025, optim0_lr0=4.839e-05, train_time=8.148
[W pthreadpool-cpp.cc:90] Warning: Leaking Caffe2 thread-pool after fork. (function pthreadpool)
```

At the same time, another conformer_fastspeech2_tacotron2 training on the same data but with `g2p=pyopenjtalk_accent_with_pause` also suspended on 14epoch.

The two training programs are still staying in the memory, but their GPU utilization have become 0%.

```
$ nvidia-smi
Sat Oct 30 17:01:12 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   71C    P0    32W /  70W |   8431MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            Off  | 00000000:00:05.0 Off |                    0 |
| N/A   68C    P0    29W /  70W |   7882MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       920      G   /usr/lib/xorg/Xorg                 95MiB |
|    0   N/A  N/A      1032      G   /usr/bin/gnome-shell                8MiB |
|    0   N/A  N/A      1849      C   python3                          8323MiB |
|    1   N/A  N/A       920      G   /usr/lib/xorg/Xorg                  4MiB |
|    1   N/A  N/A      1588      C   python3                          7875MiB |
+-----------------------------------------------------------------------------+
```

I thought it may have something to do with the warning message appears at the end of train.log:

```
[W pthreadpool-cpp.cc:90] Warning: Leaking Caffe2 thread-pool after fork. (function pthreadpool)
```

However, it seems that it is not produced by the espnet source code.

Did you ever meet such case before? What kind of reasons may cause the training suspend?

Thanks in advance.

**Comment by kan-bayashi** [comment]
It seems pytorch issue.
https://github.com/pytorch/pytorch/issues/57273

I have no idea to solve it but if you provide your environmental information in issue template, I may get some insights to solve it.

**Comment by zzxiang** [comment]
Thanks for your kindly response, Tomoki.

Here's my environmental information:

**Basic environments:**

- OS information: Linux 5.11.0-1021-gcp #23~20.04.1-Ubuntu SMP Fri Oct 1 19:04:32 UTC 2021 x86_64
- python version: `3.8.10 (default, Jun  4 2021, 15:09:15)  [GCC 7.5.0]`
- espnet version: `espnet 0.10.3`
- pytorch version: `pytorch 1.9.0`
- Git hash: `9e1b9d533593ee632efd44d15448a3b4a6e44678`
  - Commit date: `Mon Oct 11 21:37:00 2021 +0900`

**Environments from `torch.utils.collect_env`:**

```
Collecting environment information...
PyTorch version: 1.9.0
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 8.4.0-3ubuntu2) 8.4.0
Clang version: Could not collect
CMake version: version 3.19.6
Libc version: glibc-2.31

Python version: 3.8 (64-bit runtime)
Python platform: Linux-5.11.0-1021-gcp-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 11.1.74
GPU models and configuration: 
GPU 0: Tesla T4
GPU 1: Tesla T4

Nvidia driver version: 460.91.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.0
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.9.0
[pip3] torch-complex==0.2.1
[pip3] torch-optimizer==0.1.0
[pip3] torchaudio==0.9.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.1.1               h6406543_8    conda-forge
[conda] mkl                       2021.2.0           h06a4308_296  
[conda] numpy                     1.21.0                   pypi_0    pypi
[conda] pytorch                   1.9.0           py3.8_cuda11.1_cudnn8.0.5_0    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] pytorch-wpe               0.0.1                    pypi_0    pypi
[conda] torch-complex             0.2.1                    pypi_0    pypi
[conda] torch-optimizer           0.1.0                    pypi_0    pypi
[conda] torchaudio                0.9.0                    pypi_0    pypi
```

It might be a problem that the cudatoolkit library and nvcc are 11.1.1, but `nvidia-smi` shows that the CUDA version is 11.2. I'm not sure about that.

**Comment by zzxiang** [comment]
It seems that pytorch issue is indeed not the reason. The same warning message also appears in those training which succeeded.

**Comment by kan-bayashi** [comment]
Hmm. It seems your env is fine.
nvidia-smi shows the maximum supported cuda version so it should be no problem.
For now, resuming from the hung epoch is workaround from myside.

**Comment by zzxiang** [comment]
I tried restarting the machine and resuming from the hung epoch but didn't succeed. It still hung at the same epoch.

Finally I deleted the experiment folder (tts1/exp/...) and restarted from 1 epoch. Also, this time I only run one training process and use only one GPU.  Now it has been moving forward to 75epoch.

I'm not sure about it, but it might be a pytorch or CUDA issue which happens on some specific computing pattern with some specific variable values...

Thank you all the same.
